### ISSUE TITLE
Swift LLDB requires Python 3.8, but Jammy has Python 3.10. Cannot start/stop a debug sesion

### ISSUE THREAD

**Issue by Zingam** [swift-server/vscode-swift#318]
**Describe the bug**

**To Reproduce**
1. Created a simple "Hello World!" project
2. Build
3. Attempt to debug F5

I receive:

```json5
configuration: {
  type: 'lldb',
  request: 'launch',
  name: 'Debug SwiftTest',
  program: '${workspaceFolder:SwiftTest}/.build/debug/SwiftTest',
  args: [],
  cwd: '${workspaceFolder:SwiftTest}',
  preLaunchTask: 'swift: Build Debug SwiftTest',
  __configurationTarget: 5,
  relativePathBase: '/home/bti/Projects/SwiftTest'
}
Error: "\"libpython3.8.so.1.0: cannot open shared object file: No such file or directory\""
Debug adapter exit code=1, signal=null.
```

`libpython3.8` seems to be installed:

$ sudo apt-get install libpython3.8
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libpython3.8-stdlib' for regex 'libpython3.8'
Note, selecting 'libpython3.8-dev' for regex 'libpython3.8'
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I'm unable to debug but also the toolbar with the debug buttons cannot be close:

![image](https://user-images.githubusercontent.com/3337419/170824952-c9e03217-1de0-4b97-843d-716bcd0ba68f.png)

![image](https://user-images.githubusercontent.com/3337419/170826362-fc401f8e-916a-4d05-930e-6e31ff290c4c.png)

Clicking on stop doesn't do anything.

**Expected behavior**

Clicking on stop should "stop" the debugging and end the debugging session.

**Environment**
 - OS: [e.g. Ubuntu 20.04]
   Ubuntu 22.04
 - Swift version (output of `swift --version`)
   Swift version 5.7-dev (LLVM 9fde71b269b42b8, Swift ddc46e1f5bd885d)
   Target: x86_64-unknown-linux-gnu
 - Visual Studio Code version:
   1.67.2
 - vscode-swift version:
   v0.5.3

**Additional context**

```Swift
// swift-tools-version:5.7
import PackageDescription

let package = Package(
    name: "SwiftTest",
    products: [
        .executable(name: "SwiftTest", targets: ["SwiftTest"])
    ],
    targets: [
        .target(
            name: "SwiftTest"
        )
    ]
)
```

So basically I have two issues:

- starting a debug session fails (maybe it's my fault)
- stopping a debug session fails - this looks like a bug

**Comment by Zingam** [comment]
More info:

If I type

```sh
$ lldb
lldb: error while loading shared libraries: libpython3.8.so.1.0: cannot open shared object file: No such file or directory

```

So I used Synaptic to install `lldb` package.

Invoking lldb from the command line still gives me the above error.

Then I remebered that on the first run I got a message, which created a option in `settings.json` pointing to `liblldb.so` or something. I don't remember the exact option or library. I deleted it and now it appears to work.

**Comment by Zingam** [comment]
I got thsi message again. It's from CodeLLDB apparently: 

![image](https://user-images.githubusercontent.com/3337419/170827800-1aa50fdc-2cf0-43ac-94e3-10f39c864129.png)

```json5
{
    "lldb.library": "/home/hristo/SDKs/Swift/swift-5.7-DEVELOPMENT-SNAPSHOT-2022-05-18-a-ubuntu20.04/usr/lib/liblldb.so"
}
```

**Comment by adam-fowler** [comment]
Hi,

The swift version of lldb seems to be a common failure point. For the moment remove the `lldb.library` setting and ignore the "CodeLLDB requires..." dialog at startup. Because this has become such a common failure point I'm going to add a setting to not update the CodeLLDB library setting.

Last time I checked this was all working on my Ubuntu setup, I will verify this isn't an issue with swift 5.7 dev builds

**Comment by Zingam** [comment]
I created a link for the missing python library. So it works now for me. 

```sh
$sudo ln -s /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0 /usr/lib/x86_64-linux-gnu/libpython3.8.so.1.0
```

The lldb requires libpython3.8.so.1.0. On Ubuntu 22.04 it is 3.10 It's another matter why can't Swift be distributed as a universal binary for all Linux distributions.

If I use the system default lldb (the one from the Ubuntu package) I don't get the pretty printers I think. 

What apparently is an actual issue is that that the extension is hanging according to this comment: 

https://github.com/vadimcn/vscode-lldb/issues/709#issuecomment-1140295138

**Comment by adam-fowler** [comment]
Swift doesn't officially support Ubuntu 22.04 yet. It is worthwhile adding a bug to https://github.com/apple/swift telling them that swift lldb requires python 3.8 but Ubuntu 22.04 default version of python is 3.10.

Otherwise have you used the swift installs from https://swiftlang.xyz. There is a forum post about it here https://forums.swift.org/t/new-swift-package-repository-for-ubuntu-debian-linux-distributions/50412 . I think they have a Jammy install.

**Comment by adam-fowler** [comment]
I will try and catch the situation where lldb fails because it can't find a dynamic library and disable the swift lldb at that point.

**Comment by Zingam** [comment]
> Swift doesn't officially support Ubuntu 22.04 yet. It is worthwhile adding a bug to https://github.com/apple/swift telling them that swift lldb requires python 3.8 but Ubuntu 22.04 default version of python is 3.10.
> 
> Otherwise have you used the swift installs from https://swiftlang.xyz. There is a forum post about it here https://forums.swift.org/t/new-swift-package-repository-for-ubuntu-debian-linux-distributions/50412 . I think they have a Jammy install.

Thank you. I didn't know about that. I only installed Swift on Linux a few days ago. The extension feels very promising. I use VSCode for C/C++ development on macOS/Linux daily and now I can do cross-platform Swift too, which is awesome.

**Comment by adam-fowler** [comment]
@Zingam Swift just release Trunk Dev snapshots for Ubuntu 22.04 https://www.swift.org/download/

**Comment by eucharistia** [comment]
This issue has reappeared for me with Swift 6.0.2 on Ubuntu 24.10, I tried updating to 6.1 but this only upped the libpython dependency to 6.10, which was closer, but Ubuntu 24.10 comes with the latest Python 3.12 so I had to run this command to get 6.1 running:

`sudo ln -s /usr/lib/x86_64-linux-gnu/libpython3.12.so.1.0 /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0`

This should work if you are missing libpython 3.8 and you have 3.10:

`sudo ln -s /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0 /usr/lib/x86_64-linux-gnu/libpython3.8.so.1.0`
